### ISSUE TITLE
C++ front-end cannot parse names that resemble integer suffixes

### ISSUE THREAD

**Issue by wandernauta** [utwente-fmt/vercors#1294]
When verifying a C++ program, any variable, function or namespace name that looks like `u`, `ul`, `ull`, `l`, `lu`, `llu` (case insensitive) cannot be parsed:

```cpp
namespace u {
}
```

It appears that the token is recognized as an integer suffix instead of an identifier.

<details>
<pre>
======================================
At /tmp/vercors-interpreted-10485601513190727955.ipp
--------------------------------------
    6  # 1 "<built-in>" 2
    7  # 1 "<stdin>" 2
                [-
    8  namespace u {
                 -]
    9  }
   10
--------------------------------------
[1/1] Parsing failed: no viable alternative at input 'namespaceu'
======================================
[INFO] Done: VerCors (at 16:23:50, duration: 00:00:00)
</pre>
</details>

The problem seems specific to the C++ front-end and does not occur with C.

Version: 077f8b9842b25df833bd24e4a40cea43c34e57f8 (dev branch).

This issue was found by fuzzing.

**Comment by superaxander** [comment]
Yes I think you're right. Compared to https://github.com/antlr/grammars-v4/blob/master/cpp/CPP14Lexer.g4 the order between the Identifier and IntegerSuffix tokens is reversed which is probably the cause of the issue
